src/v/kafka/server/group_manager.cc

```diff
@@ -386,6 +386,34 @@ recovery_batch_consumer::operator()(model::record_batch batch) {
         auto [group_it, _] = st.groups.try_emplace(val.group_id, group_stm());
         group_it->second.update_prepared(batch.last_offset(), val);
 
+        return ss::make_ready_future<ss::stop_iteration>(
+          ss::stop_iteration::no);
+    } else if (batch.header().type == cluster::group_commit_tx_batch_type) {
+        vassert(
+          batch.record_count() == 1,
+          "commit batch must contain a single record");
```

Comment:
can you add more context to this assertion. this is the last time we have before we exit abruptly. may printing some large initial state, but say a customer sees this crash, we need to try and deduce, how this could be possible. 

vasserts can print a lot of stuff :)

Reply:
there is no context in which record_count isn't one, when we create a group_commit_tx_batch_type it is always contains a single record